### ISSUE TITLE
Generate d.ts from exported object literals

### ISSUE THREAD

**Issue by EisenbergEffect** [YoloDev/babel-dts-generator#8]
If a module exports an object literal, it ends up being exported as a single symbol of type `any`. It would be nice if the generator could handle this scenario by outputing the actual shape of the object in the d.ts.

**Comment by cmichaelgraham** [comment]
I'd like to take a shot at this.  I'll start down the path and see if I can implement it.  @Alxandr - will submit for your review and feedback when ready :+1:

**Comment by Alxandr** [comment]
Please provide a sample.

**Comment by cmichaelgraham** [comment]
sample: 

when an object literal is defined like [this](https://github.com/aurelia/metadata/blob/master/src/metadata.js#L76-L112), 

``` javascript
/**
* Provides helpers for working with metadata.
*
* @class Metadata
* @static
*/
export var Metadata = {
  global: theGlobal,
  resource:'aurelia:resource',
  paramTypes:'design:paramtypes',
  properties:'design:properties',
  get(metadataKey:string, target:Function, targetKey:string){
    if(!target){
      return undefined;
    }

    let result = Metadata.getOwn(metadataKey, target, targetKey);
    return result === undefined ? Metadata.get(metadataKey, Object.getPrototypeOf(target), targetKey) : result;
  },
  getOwn(metadataKey:string, target:Function, targetKey:string){
    if(!target){
      return undefined;
    }

    if(target.hasOwnProperty('decorators')){
      ensureDecorators(target);
    }

    return Reflect.getOwnMetadata(metadataKey, target, targetKey);
  },
  define(metadataKey:string, metadataValue:string, target:Function, targetKey:string){
    Reflect.defineMetadata(metadataKey, metadataValue, target, targetKey);
  },
  getOrCreateOwn(metadataKey:string, Type:Function, target:string, targetKey:string){
    let result = Metadata.getOwn(metadataKey, target, targetKey);

    if(result === undefined){
      result = new Type();
      Reflect.defineMetadata(metadataKey, result, target, targetKey);
    }

    return result;
  }
```

it gets [this](https://github.com/aurelia/metadata/blob/master/dist/amd/aurelia-metadata.d.ts#L10) in the .d.ts file

``` javascript
 export var Metadata: any;
```

**Comment by Alxandr** [comment]
Yeah, as we discussed on Gitter, this is basically type inference, which is entirely out of scope unfortunately. The amount of things that would be needed to get this to work properly is not funny. The way you solve this is either by creating a interface, or typing it explicitly as a object type like so:

``` typescript
export let Metadata: {resource: string, paramTypes: string} = yourObjectHere;
```

One of the things to note from this is a design principle of this plugin; anything that is not explicitly typed, will not get typed (or just default to `any`). Closing this as wont fix.
